sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -441,13 +441,20 @@ class SymmetricHashJoinStateManager(
     val keyToNumValuesMetrics = keyToNumValues.metrics
     val keyWithIndexToValueMetrics = keyWithIndexToValue.metrics
     def newDesc(desc: String): String = s"${joinSide.toString.toUpperCase(Locale.ROOT)}: $desc"
+    def newDescNotes: String = joinSide.toString.toUpperCase(Locale.ROOT)
 
     StateStoreMetrics(
       keyWithIndexToValueMetrics.numKeys,       // represent each buffered row only once
       keyToNumValuesMetrics.memoryUsedBytes + keyWithIndexToValueMetrics.memoryUsedBytes,
       keyWithIndexToValueMetrics.customMetrics.map {
         case (metric, value) => (metric.withNewDesc(desc = newDesc(metric.desc)), value)
-      }
+      },
+      // We want to collect instance metrics from both state stores
+      (keyWithIndexToValueMetrics.instanceMetrics ++ keyToNumValuesMetrics.instanceMetrics)
+        .collect {
+          case (metric, value) =>
+            (metric.withNewDescNotes(descNotes = newDescNotes), value)
```

Comment:
stateStoreName contains the side information, so if the metric name contains state store name, we don't need the note.

Reply:
Gotcha, that makes sense! Will take that out in that case